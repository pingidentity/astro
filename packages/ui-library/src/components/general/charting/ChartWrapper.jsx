import React from "react";
import PropTypes from "prop-types";
import Spinner from "../../general/PageSpinner";
import Text, { textTypes } from "../../general/Text";
import Padding, { sizes } from "../../layout/Padding";


const BASE_CLASS_NAME = "chart-wrapper";
const BASE_SPACING = sizes.SM;

function ChartWrapper({
    legend,
    title,
    chart,
    controls,
    loadingMessage,
    message,
}) {
    return (
        <div className={BASE_CLASS_NAME}>
            {title}
            <Padding bottom={BASE_SPACING} />
            {legend}
            <Padding bottom={BASE_SPACING} />
            <div className={`${BASE_CLASS_NAME}__chart`}>
                {loadingMessage ? (
                    <div className={`${BASE_CLASS_NAME}__loader`}>
                        <Spinner small show={true} />
                        <Text type={textTypes.SECTIONTITLE}>{loadingMessage}</Text>
                    </div>
                ): null}
                {
                    message !== undefined ? (
                        <div className={`${BASE_CLASS_NAME}__message`}>
                            <Text type={textTypes.SECTIONTITLE}>{message}</Text>
                        </div>
                    ) : null
                }
                {chart}
            </div>
            <Padding bottom={BASE_SPACING} />
            {controls}
        </div>
    );
}

ChartWrapper.propTypes = {
    message: PropTypes.node,
    legend: PropTypes.node,
    controls: PropTypes.node,
    loadingMessage: PropTypes.node,
    title: PropTypes.node,
};

export default ChartWrapper;
