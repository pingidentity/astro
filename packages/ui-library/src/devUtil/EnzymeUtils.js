import { mount } from "enzyme";


export const snapshotDataIds = node => {
    const elements = node.find("[data-id]").hostNodes(); // Find only actual HTML elements

    const ids = elements.map(element => element.prop("data-id")).filter(item => item !== "");
    expect(ids).toMatchSnapshot();
};

export const mountSnapshotDataIds = node => snapshotDataIds(mount(node));
