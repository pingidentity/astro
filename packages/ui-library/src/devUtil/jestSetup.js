
// This is needed for testing the modal and DetailsTooltip component.
global.getSelection = jest.fn();
global.getSelection.mockReturnValue({
    toString: () => "",
});
global.document.createRange = () => ({
    setStart: () => {},
    setEnd: () => {},
    commonAncestorContainer: {
        nodeName: "BODY",
        ownerDocument: document,
    },
});
