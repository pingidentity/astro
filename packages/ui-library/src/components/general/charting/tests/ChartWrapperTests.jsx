import React from "react";
import { shallow } from "enzyme";
import { mountSnapshotDataIds } from "../../../../devUtil/EnzymeUtils";
import ChartWrapper from "../ChartWrapper";

describe("ChartWrapper", () => {
    const defaultProps = {
        "data-id": "test-wrapper",

    };
    const getComponent = props => shallow(<ChartWrapper {...defaultProps} {...props} />);

    it("renders the component", () => {
        const component = getComponent();
        expect(component.exists()).toEqual(true);
    });

    it("data-id's don't change", () => {
        mountSnapshotDataIds(
            <ChartWrapper />
        );
    });

    it("shows loading message if loadingMessage prop is passed in", () => {
        const component = getComponent({
            loadingMessage: "loading"
        });

        expect(component.find(".chart-wrapper__loader").exists()).toEqual(true);
    });

    it("does not show loading message if loadingMessage prop is not passed in", () => {
        const component = getComponent();

        expect(component.find(".chart-wrapper__loader").exists()).toEqual(false);
    });

    it("shows message if message prop is passed in", () => {
        const component = getComponent({
            message: "loading"
        });

        expect(component.find(".chart-wrapper__message").exists()).toEqual(true);
    });

    it("does not show loading message if message prop is not passed in", () => {
        const component = getComponent();

        expect(component.find(".chart-wrapper__message").exists()).toEqual(false);
    });
});