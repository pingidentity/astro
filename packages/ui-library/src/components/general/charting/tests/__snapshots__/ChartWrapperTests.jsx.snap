// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartWrapper data-id's don't change 1`] = `
Array [
  "padding-component",
  "padding-component",
  "padding-component",
]
`;
